### PR TITLE
Docker images for ArgusClient, ArgusWeb, and ArgusWebServices

### PR DESCRIPTION
**Pull request by rmelick**
1st draft of creating docker images for argus client, argus web, and argus web services.  There are still some open questions though:

- Build process
  - Do we want the docker images built during the argus build process, or built using a docker hub automated build (https://docs.docker.com/docker-hub/builds/)?  To do the automated build, the Dockerfiles would live in a separate repository, and would have to use publicly available files.  This would work fine for the argus-client, since it is available in maven central, but the argusWeb node application and argusWebServices war are not publicly available anywhere as far as I know.
- Docker hub project
  - To match the ownership of the github repository, we should probably create a salesforce organization in docker hub (https://docs.docker.com/docker-hub/orgs/)
- Separating the configuration from the build process.  
  - For ArgusWebServices, we can already override CATALINA_OPTS to set argus.config.public.location system property.  
  - For ArgusClient, we would need to improve the client.sh to support passing additional arguments to the java command.  
  - For ArgusWeb, I don't know what would need to be done, I have very little experience with node applications
- docker plugin
  - I chose the spotify docker-maven-plugin (https://github.com/spotify/docker-maven-plugin), because it seemed more focused on just building the images.  The other option seemed to be the fabric8io plugin (https://github.com/fabric8io/docker-maven-plugin).  If we use the automated docker build, we would not have to worry about this.
- docker tags
  - I set it up so that the images are tagged with the version number used during the build.  The plugin seems to also be automatically tagging them with the tag "latest".

I'm also thinking it would be useful to add two different docker-compose files to show how the images would actually be deployed.  One file would be basic (using derby for the db and no ldap), and the other would be complete with a postgres, ldap, etc.  It's possible these could be used during integration testing as well.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -128,6 +128,7 @@
         <maven.compiler.target>1.8</maven.compiler.target>
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <maven.build.timestamp.format>yyyy-MM-dd HH:mm</maven.build.timestamp.format>
+        <dockerImages.base.name>rmelick</dockerImages.base.name>
```
Comment (cnardi-dev) [inline pom.xml]: This looks awesome! Thank you so much for offering this help to us!

File: ArgusDocker/simple/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    environment:
+      - "CATALINA_OPTS=-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=5005 -Dargus.config.public.location=/usr/argus/argus.properties"
+  argus-web:
+    image: rmelick/argus-web
+    hostname: argus-web
+    container_name: argus-web
+    ports:
+      - "8082:8000"
+    volumes:
+      - ./config.js:/usr/argus/argusWeb/app/js/config.js:ro
```
Comment (rmelick) [inline ArgusDocker/simple/docker-compose.yml]: This seemed like the correct way to set the config of the node app.  Is that correct?

**Comment by aertoria** [comment]
@cksassy
@AmeyRuikar

File: ArgusDocker/simple/docker-compose.yml (hunk 2 above)
Comment (saaja-sfdc) [inline ArgusDocker/simple/docker-compose.yml]: correct

**Comment by saaja-sfdc** [comment]
@rmelick I'm checking with our OSS team about salesforce org on docker hub.  I'm going to wait 24h for an update before merging this.

**Comment by rmelick** [comment]
@saaja-sfdc did you ever hear back from the OSS team?

**Comment by saaja-sfdc** [comment]
I did. They are going through the process of setting up an official repo.  We can update the image info when it becomes available.

**Comment by rmelick** [comment]
@saaja-sfdc have they set up the official repo yet?

File: pom.xml (hunk 1 above)
Comment (saaja-sfdc) [inline pom.xml]: @rmelick Assuming this latest commit passes testing, I'm going to go ahead and merge this.  We still have no official SF docker repo, nor a clear indication of when that will occur.  We can revisit the base name when it is in place.

File: ArgusDocker/persistent/conf/argus-web-services/tomcat-users.xml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	-->
+	<!--
+	  <role rolename="tomcat"/>
+	  <role rolename="role1"/>
+	  <user username="tomcat" password="<must-be-changed>" roles="tomcat"/>
+	  <user username="both" password="<must-be-changed>" roles="tomcat,role1"/>
+	  <user username="role1" password="<must-be-changed>" roles="role1"/>
+	-->
+	<role rolename="manager-gui"/>
+	<user username="argus" password="g3h31m" roles="manager-gui"/>
```
Comment (saaja-sfdc) [inline ArgusDocker/persistent/conf/argus-web-services/tomcat-users.xml]: @rmelick does it make sense to make the password 'changeme' or similar for the image?

File: ArgusDocker/persistent/conf/config.js [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. */
+angular.module("argus.urlConfig", [])
+.constant('CONFIG', {
+    version: '2.2.0-SNAPSHOT',
```
Comment (saaja-sfdc) [inline ArgusDocker/persistent/conf/config.js]: does this need an update to version?

File: ArgusDocker/simple/argus-web-services/tomcat-users.xml [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+	-->
+	<!--
+	  <role rolename="tomcat"/>
+	  <role rolename="role1"/>
+	  <user username="tomcat" password="<must-be-changed>" roles="tomcat"/>
+	  <user username="both" password="<must-be-changed>" roles="tomcat,role1"/>
+	  <user username="role1" password="<must-be-changed>" roles="role1"/>
+	-->
+	<role rolename="manager-gui"/>
+	<user username="argus" password="g3h31m" roles="manager-gui"/>
```
Comment (saaja-sfdc) [inline ArgusDocker/simple/argus-web-services/tomcat-users.xml]: Perhaps 'changeme' or similar for password.

File: ArgusDocker/simple/config.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+ * PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+ * LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+ * EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. */
+angular.module("argus.urlConfig", [])
+.constant('CONFIG', {
+    version: '2.2.0-SNAPSHOT',
```
Comment (saaja-sfdc) [inline ArgusDocker/simple/config.js]: Version update needed?

File: ArgusDocker/persistent/conf/argus-web-services/tomcat-users.xml (hunk 3 above)
Comment (rmelick) [inline ArgusDocker/persistent/conf/argus-web-services/tomcat-users.xml]: yeah, makes sense

File: ArgusDocker/persistent/conf/config.js (hunk 4 above)
Comment (rmelick) [inline ArgusDocker/persistent/conf/config.js]: yep

File: ArgusDocker/simple/argus-web-services/tomcat-users.xml (hunk 5 above)
Comment (rmelick) [inline ArgusDocker/simple/argus-web-services/tomcat-users.xml]: yep

File: ArgusDocker/simple/config.js (hunk 6 above)
Comment (rmelick) [inline ArgusDocker/simple/config.js]: yep

**Pull request merged**